net/net.go

```diff
@@ -16,7 +16,128 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRs parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRs(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, 0, len(cidrsString))
+	for _, cidrString := range cidrsString {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cidr value:%q with error:%v", cidrString, err)
+		}
+		cidrs = append(cidrs, cidr)
+	}
+	return cidrs, nil
+}
+
+// IsDualStackIPs returns if a slice of ips is:
+// - all are valid ips
+// - at least one ip from each family (v4 or v6)
+// - an error is returned if length is less than 2
+// - an error is returned if duplicate ips are found
+func IsDualStackIPs(ips []net.IP) (bool, error) {
+	if len(ips) < 2 {
```

Comment:
Does this need to be an error?  A slice of one element is not dual-stack, why is it an error?

Reply:
if we treated `[validv4, validv6, invalid]` as an error case, then we should treat `[validv4]` as an error case. Since we always look at the entire slice, no?

Reply:
I don't follow.

[validv4] is valid but not dual-stack (not error)

[validv4, validv6, invalid] is not valid (error)

Reply:
removed the len < 2check